spec/commands/delete_repos_spec.rb

```diff
@@ -0,0 +1,72 @@
+require 'spec_helper'
+
+describe 'delete_repos' do
+  include CommandHelpers
+
+  def common_test(create_as_public)
+    request_stubs = []
+
+    request_stubs << stub_get_json('https://testteacher:abc123@api.github.com/orgs/testorg',
+      login: 'testorg',
+      url: 'https://api.github.com/orgs/testorg'
+    )
+    request_stubs << stub_get_json('https://testteacher:abc123@api.github.com/orgs/testorg/teams?per_page=100', student_teams)
+    student_usernames.each do |username|
+      # Check for the repos existing already
+      stub_request(:get, "https://testteacher:abc123@api.github.com/repos/testorg/#{username}-testrepo").
+        to_return(status: 404)
+    end
+
+    student_usernames.each do |username|
+      # create the repo for test
+      team_id = 0
+      student_teams.each do |st|
+        if st[:name].eql?(username)
+          team_id = st[:id]
+        end
+      end
+      stub_request(:post, "https://testteacher:abc123@api.github.com/orgs/testorg/repos").
+        with(body: {
+          description: "testrepo created for #{username}",
+          private: !create_as_public,
+          has_issues: true,
+          has_wiki: false,
+          has_downloads: false,
+          team_id: team_id,
+          name: "#{username}-testrepo"
+        }.to_json)
+    end
+
+    teachers_pet(:create_repos,
+      repository: 'testrepo',
+      organization: 'testorg',
+      public: create_as_public,
+
+      students: students_list_fixture_path,
+
+      username: 'testteacher',
+      password: 'abc123'
+    )
+    teachers_pet(:delete_repos,
+      repository: 'testrepo',
+      organization: 'testorg',
+
+      students: students_list_fixture_path,
+
+      username: 'testteacher',
+      password: 'abc123'
+    )
+
+    request_stubs.each do |request_stub|
+      expect(request_stub).to have_been_requested.twice
```

Comment:
I think I got problem here, I copied and modified this file from `lib/teachers_pet/actions/create_repos.rb`.

If I set `have_been_requested.` to once, I'll get
>      Failure/Error: expect(request_stub).to have_been_requested.once
> The request GET https://testteacher:abc123@api.github.com/orgs/testorg/teams?per_page=100 was expected to execute 2 times but it executed 1 time

But if I set it to twice, I'll get 
>      Failure/Error: expect(request_stub).to have_been_requested.once
       The request GET https://testteacher:abc123@api.github.com/orgs/testorg was expected to execute 1 time but it executed 2 times